docs/platforms/native/advanced-usage/backend-tradeoffs/index.mdx

```diff
@@ -0,0 +1,69 @@
+---
+title: Backend Tradeoffs
+description: "How to choose the right crash backend in the Native SDK."
+sidebar_order: 1000
+---
+The Native SDK lets users decide at compile-time between three crash backends:
+
+* `crashpad`
+* `breakpad`
+* `inproc`
+
+Currently, `crashpad` is the default on all desktop platforms because it
+
+* has an external `handler` process that allows for external snapshots and sending crash reports immediately (instead of on the next successful start of your application)
+* is the primary target for extension compared to upstream, including
+  * client-side stack traces
+  * attachment handling
```

Comment:
Does this indicate we don't attachments using other backends?
I believe crashpad only allows specifying attachments during `init`, is that right? That can be seeing as a limitation. And technically we should be able to support attachments on the other backends (not by the backend themselves, but by the SDK. Hold on to the paths, on restart if the files are there, send them. No?)

Reply:
> Does this indicate we don't have attachments using other backends?

No, it was meant as one of the extensions (this is a nested list) we implemented to allow _any_ attachments with crashpad (because upstream is/was incomplete).

Yes, all other backends allow attachments, and we could easily provide dynamic attachments for those. As such, the attachment story for Crashpad isn't an upside.

However, this list was meant to include things we implemented locally in our fork to provide even that level of capability. We can remove it if it makes the comparison between backends more blurry.

Reply:
Please keep it, even feel free to expand with more context such as what you wrote in this thread. If possible to make it succinct enough to fit in there.

All this stuff you're explaining to us here is very helpful to know. I feel more prepared when talking to customers about trade offs, and what level of maintenance is involved (to some degree, just good to know what we changed in the fork)